ou_dedetai/app.py

```diff
@@ -0,0 +1,261 @@
+
+import abc
+import logging
+import os
+from pathlib import Path
+import sys
+import threading
+from typing import Callable, NoReturn, Optional
+
+from ou_dedetai import constants
+from ou_dedetai.constants import (
+    PROMPT_ANSWER_COME_AGAIN,
+    PROMPT_OPTION_DIRECTORY,
+    PROMPT_OPTION_FILE
+)
+
+
+class App(abc.ABC):
+    # FIXME: consider weighting install steps. Different steps take different lengths
```

Comment:
This would be a good improvement; alternatively, split up long steps into smaller steps.